src/controllerclientimpl.cpp

```diff
@@ -190,6 +190,34 @@ ControllerClientImpl::ControllerClientImpl(const std::string& usernamepassword,
     // csrftoken can be any non-empty string
     _csrfmiddlewaretoken = "csrftoken";
     std::string cookie = "Set-Cookie: csrftoken=" + _csrfmiddlewaretoken;
+#if CURL_AT_LEAST_VERSION(7,60,0)
+    // with https://github.com/curl/curl/commit/b8d5036ec9b702d6392c97a6fc2e141d6c7cce1f, setting domain param to cookie is required.
+    if(_baseuri.find('/') == _baseuri.size()-1) {
+        // _baseuri should be hostname with trailing slash
+        cookie += "; domain=";
+        cookie += _baseuri.substr(0,_baseuri.size()-1);
+    } else {
+        CURLUcode rc;
+        CURLU *url = curl_url();
+        BOOST_SCOPE_EXIT_ALL(&url) {
+            curl_url_cleanup(url);
+        };
+        rc = curl_url_set(url, CURLUPART_URL, _baseuri.c_str(), 0);
```

Comment:
We have `CHECKCURLCODE` and we check against `CURLE_OK`, I think this `!rc` check is way too cryptic.

Reply:
If we cannot parse the url, maybe there is no point of continuing.

Reply:
umm, sure

Reply:
sorry Ziyan, I got this

```
/home/mujin/mujin/checkoutroot/controllerclientcpp/src/controllerclientimpl.cpp:205:35: error: cannot convert ‘CURLUcode’ to ‘CURLcode’
  205 |         CHECKCURLCODE(curl_url_set(url, CURLUPART_URL, _baseuri.c_str(), 0), "cannot parse url");
  ```

Reply:
let me add new macro

Reply:
sorry again, curl_url_strerror is available only from curl 7.80

Reply:
eh, done dirty hack anyway, please decide if this commit should be reverted or not

Reply:
@ziyan sure, ok, thank you for accepting this state.